src/flint/fmpq.jl

```diff
@@ -883,6 +883,15 @@ end
 
 (a::FlintRationalField)(b::fmpq) = b
 
+function (a::FlintIntegerRing)(b::fmpq)
+   z = fmpz()
+   ccall((:fmpq_denominator, libflint), Nothing, (Ref{fmpz}, Ref{fmpq}), z, b)
+   isone(z) || error("Denominator must be 1")
```

Comment:
Is it possible to do this without having to copy the denominator of `b`?

Reply:
oh, Tommy already asked this

Reply:
No, not officially. It's ok as it is easy to copy 1, and it is also already an improvement over the hecke version, which creates two extra garbage fmpz's.